feat: UI enhancement - Custom Titlebar

aL0NEW0LF opened the pull request:
To be honest the default titlebar just annoyed me, so i made a custom component to emulate it that can blend it with the UI and its colors.

aL0NEW0LF committed:
feat: UI enhancement - Custom Titlebar

To be honest the default titlebar just annoyed me, so i made a custom component to emulate it that can blend it with the UI and its colors

aL0NEW0LF committed:
fix: Build error fix

In the TitleBar component, this line was returning false when fullscreen is true:
return (
  !fullscreen && (
    // ...
  )
);

Since false is not a valid JSX element, we were getting an error.
To fix this, i made sure TitleBar always returns JSX.

Stormix reviewed (changes requested):
First of all, thank you for your contributions to the Athan app!

But, I think you can achieve the same behavior using the packages we already have, namely: tailwind and radix-icons. So I invite you to try and refactor your code to use these packages and please leave the permissions list as detailed as it currently is (so it doesn't cause us issues when submitting to the microsoft store).

Kind Regards, 
Anas

aL0NEW0LF committed:
fix: SideBar breaking at a certain height, dependencies, styling & permissions

Removed the heavy packages, re-done styling with tailwindcss, refactored permissions, refactored the TitleBar component to be optimal and efficient, fixed the SideBar that was breaking at a certain height (exactly 850px) when the view height is less than that, the Sidebar doesn't take 100% of the window height anymore, so i fixed the min-height at 850px.

aL0NEW0LF commented:
Here is a preview of what the UI has become.
![athan-screen-1](https://github.com/The-Embassy-Collective/athan/assets/114300398/2f7a08b7-f478-4f91-990a-45ef2011f4b7)
![athan-screen-2](https://github.com/The-Embassy-Collective/athan/assets/114300398/d2b2cdea-7a5c-48a3-8c65-6c11a48ac496)
![athan-screen-3](https://github.com/The-Embassy-Collective/athan/assets/114300398/fc196281-84c0-4afd-a4c7-4be77002273e)
![athan-screen-4](https://github.com/The-Embassy-Collective/athan/assets/114300398/aaba31fa-bfe1-4338-809d-8aa6f61ca7bd)

aL0NEW0LF committed:
refactor: changed the tailwind color classes
